references/optical_flow/train.py

```diff
@@ -209,6 +209,12 @@ def main(args):
         raise ValueError("The device must be cuda if we want to run in distributed mode using torchrun")
     device = torch.device(args.device)
 
+    if args.use_deterministic_algorithms:
```

Comment:
Hi Nicolas, I have just tried what you suggested and got error when I try to use `test-only`:
```
  File "/fsx/users/yosuamichael/conda/envs/vision-c113/lib/python3.9/site-packages/torch/nn/modules/linear.py", line 103, in forward
    return F.linear(input, self.weight, self.bias)
RuntimeError: Deterministic behavior was enabled with either `torch.use_deterministic_algorithms(True)` or `at::Context::setDeterministicAlgorithms(true)`, but this operation is not deterministic because it uses CuBLAS and you have CUDA >= 10.2. To enable deterministic behavior in this case, you must set an environment variable before running your PyTorch application: CUBLAS_WORKSPACE_CONFIG=:4096:8 or CUBLAS_WORKSPACE_CONFIG=:16:8. For more information, go to https://docs.nvidia.com/cuda/cublas/index.html#cublasApi_reproducibility
```
Following the error note, when I set the env var `CUBLAS_WORKSPACE_CONFIG=:4096:8` it works, but I think this will create a friction to users.

I think the main issue is that we use `torch.use_deterministic_algorithms(True)` when we set `args.use_deterministic_algorithms` and this is much more stricter than `torch.backends.cudnn.deterministic = True` when we use `args.test_only`. (see [here](https://pytorch.org/docs/stable/generated/torch.use_deterministic_algorithms.html#torch.use_deterministic_algorithms) )

Hence as of now we can only avoid duplicate of the line `torch.backends.cudnn.benchmark = False` which I think still okay to have 1 line duplicate for now. What do you think?

Reply:
Ah, fair point. I forgot that one is stricter than the other. I think the way you did it is fine then. Sorry for the noise!